services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -460,13 +460,16 @@ async def prepare_services_environment(
         )
     )
 
-    tasks = [sidecars_client.pull_service_output_ports(dynamic_sidecar_endpoint)]
+    tasks = [
+        sidecars_client.pull_user_services_images(dynamic_sidecar_endpoint),
```

Comment:
ok, so pulling images and outputs are done concurrently
Q: why not inputs as well?

Reply:
This is not straight forward to answer.

Please refer to the following issue and especially my reply at the bottom.
https://github.com/ITISFoundation/osparc-issues/issues/1523#issuecomment-2306735110